sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DeploymentStep.java

```diff
@@ -175,12 +187,22 @@ private void setTaskStatus(Protos.TaskID taskID, Status status) {
             tasks.replace(taskID, new TaskStatusPair(tasks.get(taskID).getTaskInfo(), status));
             logger.info("Status for: {} is: {}", taskID.getValue(), status);
         }
+
+        if (getStatus().equals(Status.PENDING)) {
+            prepared.set(false);
```

Comment:
setting `prepared` here before reading `prepared` in `getStatus(tasks)` looks risky.